### ISSUE TITLE
How to change paste behavior in evil mode

### ISSUE THREAD

**Issue by veverjak** [syl20bnr/spacemacs#5070]
I am coming from vim world and I used to use yank -> paste to replace text with source text multiple times. 
Currently if I do [y] copy -> [p] paste -> [p] paste multiple times I get following result: 
![animation](https://cloud.githubusercontent.com/assets/606450/13036359/bf39bfa4-d365-11e5-89f7-9937b42457fb.gif)

I would love to set default behavior to paste without remembering text that was replaced or to move this text to 2nd place and not replacing originaly yanked text. 
Shortly I would like to have result of above commands just `one` on each line.

Is there any way to do that in spacemacs?

**Comment by syl20bnr** [comment]
Your gif _is_ the default Vim behavior.
You have something in your Vim config to change the default behavior.
In default Vim you have to use the register 0 if you want the original yanked text, Evil is not different in this regard.

**Comment by veverjak** [comment]
True @syl20bnr, vim's default is the same, I've changed it and forgot that I did that :) 

Would you help me or point me how can I remap [p] to achieve this behavior?

**Comment by syl20bnr** [comment]
That's a good question and it would be useful for a lot of people, I don't know how to do it with stock Evil.

I use the paste transient state (which has been reworked in current develop branch).
In master set `dotspacemacs-enable-paste-micro-state` to t and when you press `p` several time you cycle the between the registers.
In develop set `dotspacemacs-enable-transient-sate` to t and you can press `p C-j` to get the previous yanked text.

Also if you want to replace the same text with what is currently yanked you can do `SPC v s E p` which is:
- `SPC v` expand-region, press `v` multiple times to expand the region by semantic units
- `s E` is iedit which select all the occurences and allow to edit them simultaneously
- `p` replaces all the occurences with the current yanked text.

**Comment by veverjak** [comment]
Thanks @syl20bnr for the answer but I would like to go even further. 
In Vim, I used to have `xnoremap p "_dP` - solution described here: http://vim.wikia.com/wiki/Replace_a_word_with_yanked_text

`dotspacemacs-enable-paste-micro-state` is nice but if I want to paste same text over 5 different texts, this gets annoying. 

Would you know how to remap `p` to not to overwrite latest yanked value?

**Comment by justbur** [comment]
how about this? 

```elisp
(defun evil-paste-after-from-0 ()
  (interactive)
  (let ((evil-this-register ?0))
    (call-interactively 'evil-paste-after)))

(define-key evil-visual-state-map "p" 'evil-paste-after-from-0)
```

**Comment by veverjak** [comment]
This is exactly it, thanks a lot @justbur

**Comment by syl20bnr** [comment]
Good candidate for a FAQ entry :-)

**Comment by growlove** [comment]
@syl20bnr I'm attempting to use `SPC v s E p` but `s E` is surrounding the select with `E` rather than triggering iedit as you said. Am I doing something wrong?

**Comment by nixmaniack** [comment]
@growlove The second step should be `SPC s E` instead of `s E`

**Comment by growlove** [comment]
@nixmaniack That worked perfectly. tyvm!

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.

**Comment by mc10** [comment]
FYI the suggestion above does not exactly do what I wanted to; it changes the behavior of paste so that cutted text (via `d`) no longer gets pasted. What I really wanted to do was to prevent Emacs from adding the newly-selected text to the kill buffer. You can do this with the following:

    (setq-default evil-kill-on-visual-paste nil)
